src/test/java/com/artipie/RepoPerms.java

```diff
@@ -0,0 +1,148 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.amihaiemil.eoyaml.YamlMappingBuilder;
+import com.amihaiemil.eoyaml.YamlNode;
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Class for generating repo permissions.
+ * @since 0.10
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class RepoPerms {
+    /**
+     * Permissions section name.
+     */
+    private static final String PERMISSIONS = "permissions";
+
+    /**
+     * Collection with user permissions.
+     */
+    private final Collection<RepoPermissions.UserPermission> usersperms;
+
+    /**
+     * Ctor.
+     */
+    public RepoPerms() {
+        this(Collections.emptyList());
+    }
+
+    /**
+     * Ctor.
+     * @param userperm Permission for a single user
+     */
+    public RepoPerms(final RepoPermissions.UserPermission userperm) {
+        this(Collections.singleton(userperm));
+    }
+
+    /**
+     * Primary ctor.
+     * @param usersperms Collection with user permissions
+     */
+    public RepoPerms(final Collection<RepoPermissions.UserPermission> usersperms) {
+        this.usersperms = usersperms;
+    }
+
+    /**
+     * Save to the storage repo with map with users and permissions.
+     * @param storage Storage
+     * @param repo Repo
+     */
+    public void saveSettings(final Storage storage, final String repo) {
+        storage.save(
+            new Key.From(String.format("%s.yaml", repo)),
+            new Content.From(
+                Yaml.createYamlMappingBuilder().add(
+                    "repo",
+                    Yaml.createYamlMappingBuilder()
+                        .add("type", "any")
+                        .add(RepoPerms.PERMISSIONS, this.permsYaml()).build()
+                ).build().toString().getBytes(StandardCharsets.UTF_8)
+            )
+        ).join();
+    }
+
+    /**
+     * Add permissions to the repo section.
+     * @param reposection YamlMapping with repo section
+     * @return YamlMapping with repo section and permissions.
+     */
+    public YamlMapping addPermissions(final YamlMapping reposection) {
+        return RepoPerms.repoSectionAsBuilder(reposection)
+            .add(RepoPerms.PERMISSIONS, this.permsYaml())
+            .build();
```

Comment:
@genryxy I think we should support `yaml` style here and modify given `reposection` by adding permissions section and return this modified yaml without building